week2 css dev

honeybeebaby opened the pull request:
working on making html look nice with css styling

honeybeebaby committed:
added a new div in index.html to help in css styling, making both sidebars be side-by-side, and added some flexbox changes to css

honeybeebaby committed:
added formatting in css and altered some html paragraph text

kaveh096 commented on personal-practice/Dice.java:
```diff
@@ -0,0 +1,23 @@
+import java.awt.*;
```
Avoid [wildcard imports.](https://google.github.io/styleguide/javaguide.html#s3.3.1-wildcard-imports)

honeybeebaby replied:
thank you for looking at and commenting on my java files, you didnt have to do that! :)

kaveh096 commented on personal-practice/Dice.java:
```diff
@@ -0,0 +1,23 @@
+import java.util.Arrays;
+import java.util.Random;
+
+public class Dice {
```
```suggestion
public final class Dice {
```

It's a good practice to make classes final.

kaveh096 commented on personal-practice/Dice.java:
```diff
@@ -0,0 +1,23 @@
+import java.awt.*;
+import java.util.Scanner;
+import java.util.concurrent.Future;
+import java.lang.reflect.Array;
+import java.util.Arrays;
+import java.util.Random;
```
sort import statements: https://google.github.io/styleguide/javaguide.html#s3.3.3-import-ordering-and-spacing

kaveh096 commented on personal-practice/Dice.java:
```diff
@@ -0,0 +1,23 @@
+    int previousRoll = -1;
+
+    public Dice() {
+
+    this.rand = new Random();
```
```suggestion
      this.rand = new Random();
```

kaveh096 commented on personal-practice/arrays.java:
```diff
@@ -0,0 +1,28 @@
+import java.lang.reflect.Array;
+import java.util.Arrays;
+
+public class arrays {
```
User [UpperCamelCase](https://google.github.io/styleguide/javaguide.html#s5.2.2-class-names) for class names and the containing java file.

kaveh096 commented on personal-practice/arrays.java:
```diff
@@ -0,0 +1,28 @@
+import java.awt.*;
+import java.util.Scanner;
+import java.lang.reflect.Array;
+import java.util.Arrays;
```
ditto

kaveh096 commented on personal-practice/blueprint.java:
```diff
@@ -0,0 +1,20 @@
+
+public class blueprint {
+
+    String name;
+    int ageInHumanYears;
+    Color furColor;
+    String size;
```
I believe these fields could be defined as `private final` which would improve encapsulation.

kaveh096 commented on personal-practice/blueprint.java:
```diff
@@ -0,0 +1,20 @@
+    Color furColor;
+    String size;
+
+    public Dog(String inputName, int inputAge, Color inputFurColor, String inputSize) {
```
```suggestion
    public blueprint(String inputName, int inputAge, Color inputFurColor, String inputSize) {
```

kaveh096 commented on personal-practice/car.java:
```diff
@@ -0,0 +1,37 @@
+import java.awt.*;
+import java.awt.Color;
+
+public class car {
```
UpperCameCase

kaveh096 commented on personal-practice/car.java:
```diff
@@ -0,0 +1,37 @@
+
+public class car {
+
+    // Data Types: 
+    // int
+    // double (float)
+    // string
+    // Color (from awt library)
+    // boolean
```
This comment block does not look useful. It looks like commented out code which is discouraged.

kaveh096 commented on personal-practice/car.java:
```diff
@@ -0,0 +1,37 @@
+    // Color (from awt library)
+    // boolean
+
+    double averageMilesPerGallon;
+    String licensePlate;
+    Color paintColor;
+    boolean areTaillightsWorking;
```
make them `private final`

kaveh096 commented on personal-practice/car.java:
```diff
@@ -0,0 +1,37 @@
+    boolean areTaillightsWorking;
+
+    public car(double inputAverageMPG, 
+    String inputLicensePlate, 
+    Color inputPaintColor, 
+    boolean inputAreTaillightsWorking) {
```
Wrapped line is indented [by 4 spaces](https://google.github.io/styleguide/javaguide.html#s4.5.2-line-wrapping-indent).

kaveh096 commented on personal-practice/car.java:
```diff
@@ -0,0 +1,37 @@
+    String inputLicensePlate, 
+    Color inputPaintColor, 
+    boolean inputAreTaillightsWorking) {
+        this.averageMilesPerGallon = inputAverageMPG;
+        this.licensePlate = inputLicensePlate;
+        this.paintColor = inputPaintColor;
+        this.areTaillightsWorking = inputAreTaillightsWorking;
```
Method block indentation by [2 spaces only](https://google.github.io/styleguide/javaguide.html#s4.2-block-indentation).

kaveh096 commented on personal-practice/car.java:
```diff
@@ -0,0 +1,37 @@
+    public double speedingUp(double currentSpeed) {
+        currentSpeed += 100;
+        return currentSpeed;
+
```
remove extra line

kaveh096 commented on personal-practice/car.java:
```diff
@@ -0,0 +1,37 @@
+        this.paintColor = newPaintColor;
+    }
+
+    public double speedingUp(double currentSpeed) {
```
```suggestion
    public static double speedingUp(double currentSpeed) {
```

kaveh096 commented on personal-practice/diceMain.java:
```diff
@@ -0,0 +1,13 @@
+        Dice dice = new Dice();
+        System.out.println(dice.roll());
+        System.out.println(dice.roll());
+        System.out.println("Previous: " + dice.previousRoll);
```
Suggestion: add a getter method instead of accessing the field directly from outside.

kaveh096 commented on personal-practice/intro.java:
```diff
@@ -0,0 +1,24 @@
+        System.out.println("Bianca");
+
+        car myCar = new car(25.5, 
+        "1BC32B", 
+        Color.BLUE, 
+        true);
+
+        car sallyCar = new car(13.9, 
+        "3D2OBN", 
+        Color.BLACK, 
+        false);
```
fix indentations.

kaveh096 commented on portfolio/src/main/webapp/index.html:
```diff
@@ -2,17 +2,91 @@
+    <div id="body-content">
+        <div id="about" class="sidebar">
+            <div id="links" class="box">
+                <p2>a crash course in me</p2>
```
did you mean `<h2>` here? I don't think i've seen `<p2>` before.

honeybeebaby replied:
chris commented the same about pN on my other pull requests, i use them to access that particular paragraph in the css file. i can see how this can get pretty confusing when your html page gets really long, i will read up on how to style the paragraphs without having to do pN

kaveh096 commented on portfolio/src/main/webapp/index.html:
```diff
@@ -2,17 +2,91 @@
+                <p3>NYU | STEP Intern | art</p3>
+                <br>
+                <a href="https://www.linkedin.com/in/bianca-macias-344a26178">LinkedIn</a>
+                <br>
```
```suggestion
                <br/>
```

ccondit replied:
I was wondering about this, as well, and found:
* https://dev.w3.org/html5/html-author/#notes,-tips-and-warnings
* https://google.github.io/styleguide/htmlcssguide.html#Document_Type

Apparently void elements should *not* have the closing /.

honeybeebaby replied:
so leave the break as it is, right?

ccondit replied:
Yes. It violates what we know about well formed xml, but is the preferred style for HTML.

kaveh096 replied:
Thanks Chris. Closing the thread.

kaveh096 reviewed (commented):
Remove `.class` files from your repo.

honeybeebaby committed:
removed personal-practice folder as it was moved to another repo

honeybeebaby committed:
changed all pN to p, make note to change css file that has selectors to pN paragraphs

ccondit reviewed (changes requested):
Before reviewing further I guess we need to merge the previous PRs?

I think this way the diff in this PR will reflect *only* the changes it is making.

honeybeebaby committed:
Merge branch 'master' into week2cssDev

biancamacias commented:
okay i merged the other two PRs, so the branch should be good now

honeybeebaby committed:
added IDs to some <p> in index.html so they can be styled in style.css

honeybeebaby committed:
merging branch because previous PR were finally merged into master, want to keep this branhc updated
Merge branch 'week2cssDev' of https://github.com/biancamacias/step-code-2020 into week2cssDev

honeybeebaby committed:
fixed styling so about me sidebar text didnt go into space

honeybeebaby committed:
fixed about-title font, accidentally added a typo which made the font smaller

kaveh096 reviewed (approved):
LGTM

ccondit commented on portfolio/src/main/webapp/index.html:
```diff
@@ -11,78 +11,81 @@
+                <br>
+                <p id="about-content">NYU | STEP Intern | art</p>
+                <br>
+                <a href="https://www.linkedin.com/in/bianca-macias-344a26178">LinkedIn</a>
```
nit - consider adding `target="_blank"` to have this link and the github link open in a new tab.

ccondit commented on portfolio/src/main/webapp/style.css:
```diff
@@ -1,10 +1,92 @@
+    padding: 20px;
+}
+
+/* paragraph with id="about-title" */
```
nit - these comments are redundant.

ccondit commented on portfolio/src/main/webapp/style.css:
```diff
@@ -1,10 +1,92 @@
-  margin-left: auto;
-  margin-right: auto;
-  width: 650px;
+    font-weight: bold;
+    margin-left: auto;
+    margin-right: auto;
+    width: 100%;
+    background-color: #9fa2b2;
+    height: 100%;
+    padding: 10px;
+    text-align: center;
```
Per [style guide](https://google.github.io/styleguide/htmlcssguide.html#Declaration_Order) can you alphabetize all of the properties within each style?

ccondit commented on portfolio/src/main/webapp/style.css:
```diff
@@ -1,10 +1,92 @@
 body {
```
Generally I think this CSS won't take effect that way you expect: tt will work if you load the HTML directly in an editor but because of how the app engine container resolves paths it won't be found when you actually run things in app engine. I think you'll need to add a leading `/` to both your style.css and your script.js when they're referenced in index.html.

honeybeebaby committed:
alphabetized style.css and deleted redundant comments

honeybeebaby committed:
added target='_blank' to links in index.html to open links in a new tab

honeybeebaby committed:
adding leading forward slash to css and js references in index.html

ccondit reviewed (approved):
LGTM 😀
